File: app/src/main/java/com/sunny/youyun/wifidirect/activity/single_trans/trans/TransPresenter.java
package com.sunny.youyun.wifidirect.activity.single_trans.trans;

import com.sunny.youyun.wifidirect.model.TransLocalFile;

import java.util.List;

/**
 * Created by Sunny on 2017/8/11 0011.
 */

class TransPresenter extends TransContract.Presenter{
    TransPresenter(TransActivity transFragment) {
        mView = transFragment;
        mModel = new TransModel(this);
    }

    @Override
    protected void start() {
        mModel.begin();
    }

    @Override
    void exit() {
        mModel.exit();
    }

    @Override
    List<TransLocalFile> getData() {
        return mModel.getData();
    }


    @Override
    void send(String[] paths) {
        mModel.send(paths);
    }

    @Override
    void update() {
        mView.update();
    }

    @Override
    void updateItem(int i, TransLocalFile transLocalFile) {
        mView.updateItem(i, transLocalFile);
    }
}

File: app/src/main/java/com/sunny/youyun/wifidirect/activity/single_trans/trans/TransPresenter.kt
package com.sunny.youyun.wifidirect.activity.single_trans.trans

import com.sunny.youyun.wifidirect.model.TransLocalFile

/**
 * Created by Sunny on 2017/8/11 0011.
 */

class TransPresenter(transFragment: TransActivity) : TransContract.Presenter() {

    override fun getData(): List<TransLocalFile> = mModel.data

    init {
        mView = transFragment
        mModel = TransModel(this)
    }

    public override fun start() {
        mModel.begin()
    }

    override fun exit() {
        mModel.exit()
    }


    override fun send(ip: String, paths: Array<String>) {
        mModel.send(ip, paths)
    }

    override fun update() {
        mView.update()
    }

    override fun updateItem(i: Int, transLocalFile: TransLocalFile) {
        mView.updateItem(i, transLocalFile)
    }
}
